proxy/proxy.go

```diff
@@ -51,19 +61,41 @@ func main() {
 
 	loggedRouter := handlers.LoggingHandler(os.Stdout, proxyRouter)
 
-	err = http.ListenAndServe(":"+c.Port, loggedRouter)
+	err = http.ListenAndServe(":" + c.Port, loggedRouter)
 	if err != nil {
 		fmt.Println(err)
 	}
 }
-
+func CallbackUrl() (string, error) {
+	var vcapApp VcapApplication
+	err := json.Unmarshal([]byte(c.VcapAppRaw), &vcapApp)
+	if err != nil {
+		return "", err
+	}
+	// we try first ssl to see if it can be used
```

Comment:
WDYT about configuring http/https via environment instead? For example, we could add a `PROTOCOL` field to `Config` that defaults to `https`. I'm also a little uncertain about how this works--is the app making a request to itself before it starts?